frontend/actions.js

```diff
@@ -98,58 +89,149 @@ export default ({setState, getState}) => {
     displayAboutOverlay: !state.displayAboutOverlay,
   })
 
-  const inputAddress = (state, e) => ({
-    sendAddress: e.target.value,
+  const clearSendAmountValidation = (state) => ({
+    sendAmount: Object.assign(state.sendAmount, {validationMsg: ''}),
+  })
+
+  const clearSendAddressValidation = (state) => ({
+    sendAddress: Object.assign(state.sendAddress, {validationMsg: ''}),
   })
 
-  // is being called debounced, thus we need to getState
-  // TODO maybe it'd be good practice to only use getState and avoid this problem?
+  const validateSendAddress = (state) => {
```

Comment:
Please don't export this as an action.

Actions should always return a state that is merged with a new one (or undefined, in which case ) - if you need helper functions for actions, like these, put them outside the exported object (maybe even into separate file ?

Same for the rest of validate fns.

Reply:
this is not exported